### PR TITLE
Add polish lang to programme.xml

### PR DESCRIPTION
**Pull request by pauldim**
I have added translations to Polish for all titles and original descriptions. There are only a few corrections of incorrect dates and data for such a huge file. In the future, I suggest separating movies from series. With more languages, this can be problematic, to process the file.

### CODE REVIEW

**Comment by scr0llbaer9000** [comment]
Seems you've often used your own movie descriptions or other sources of descriptions, instead of translating. This can be problematic, do you know the licence of these sources? Also this comes with the problem that they sometimes contain the reference to the original title, or original titles of other films. These should be avoided.

Also, you don't wanna do the Polish spoof titles?

> I suggest separating movies from series. With more languages, this can be problematic, to process the file.

It was also discussed that at least the descriptions for the other languages would go into different files (which is already possible). But either way, just separating the series won't help for long I guess. Maybe eventually, additionally, films could be separated into the different decades when they were released. On the other hand, it's kind of ridiculous that in 2024 the tools to handle these files are still a problem.

**Comment by pauldim** [comment]
Some descriptions are taken from the filmweb.pl database because your description does not fit grammatically with the inserts Last, First, Full etc. It is difficult to do this without new descriptions. You will do as you think. I don't know how to get out of this. 
As for the made-up titles, I don't think there is any point in translating them as they are made up.

**Comment by scr0llbaer9000** [comment]
> because your description does not fit grammatically with the inserts Last, First, Full

Do you mean it'll be hard to use references to actors in Polish language because their names would be spelled differently, have different endings? If you just need to append something without spacing, it will work. If the last letters of a name would have to be changed, it won't. (Maybe the script expression feature on the roadmap would help here, but I'm not sure.)

> I have noticed that some of the characters in your descriptions are also original, inconsistent with the fictional title.
Your descriptions therefore also contain real characters.

Yes, that is a known shortcoming for now, and it's also known that different contributors handled that differently, and that there are inconsistencies. And although using the real character names might be less of a legal issue than the film titles, it can also be solved with the script expression feature on the roadmap. Then roles can be defined and referenced also in programme descriptions. Using `<description_original>` was also once discussed but it's overkill for just a few character names.

> As for the made-up titles, I don't think there is any point in translating them as they are made up.

It's among the first thing users speaking your language will experience when playing the game. Eventually they will find the information what to do if they want to play with real titles, but first impressions matter too.

**Comment by pauldim** [comment]
> Do you mean it'll be hard to use references to actors in Polish language because their names would be spelled differently, have different endings, depending on the grammar? If you just need to append something without spacing, it will work. If the last letters of a name would have to be changed, it won't. (Maybe the script expression feature on the roadmap would help here, but I'm not sure.)

It is not possible in every case just to add a letter at the end.





> It's among the first thing users speaking your language will experience when playing the game. Eventually they will find the information what to do if they want to play with real titles, but first impressions matter too.

You're right. I'll translate the made-up titles in that case.

**Comment by scr0llbaer9000** [comment]
> It is not possible in every case just to add a letter at the end.

Can you point out a few examples for some names?

This will be a general problem, with translation of other things too (variables of the templates, etc).

**Comment by pauldim** [comment]
Example from file: therob.xml:
[1|Full] had been accompanied by [2|Full] as he developed and recorded his legendary radio show. Was the mass hysteria real? Here it is shown! The only feature [0|Full] did.
[2|Full] = <person id="TheRob-TowerTV-RednackRoot" first_name="Rednack" last_name="Root Jr." first_name_original="Edward D." last_name_original="Wood Jr." 
In polish: 
Orson  Welles towarzyszył  Rednack**owi** Root Jr.**owi** podczas opracowywania i nagrywania jego legendarnego programu radiowego. Czy masowa histeria była prawdziwa? Tutaj jest to pokazane! Jedyny film fabularny Fritz**a** Lang**a**.

**Comment by scr0llbaer9000** [comment]
There are several person entries with "Jr." in the dataset. Would "Jrowi" read ok? Then we can remove the last dot from all "... Jr." entries I guess, I'm almost sure it's been handled inconsistently so far and there already are some without the dot.

> Fritza Langa

This you can change in your language from `[0|Full]` to `[0|First]a [0|Last]a`.

**Comment by pauldim** [comment]
> > Fritza Langa
> 
> This you can change in your language from `[0|Full]` to `[0|First]a [0|Last]a`.

Yes. I figured out. :)

**Comment by pauldim** [comment]
> There are several person entries with "Jr." in the dataset. Would "Jrowi" read ok? Then we can remove the last dot from all "... Jr." entries I guess, I'm almost sure it's been handled inconsistently so far and there already are some without the dot.

Correct would be Juniorowi. Jr. is an abbreviation and Jrowi doesn't look right.

**Comment by nittka** [comment]
I will have a look at this PR, but due to the many additions, I cannot review every single one. The main focus will be on changes to the existing entries - Github marks 10 deletions, so there are some.

@pauldim: maybe, once your changes are merged, you could check and delete the "obsolete" branches in your fork, i.e. those branches that do not contain anything not merged here.

**Comment by pauldim** [comment]
This PR I am still updating all the time, so it will take a little longer.
As for the difference in my fork, yes I am aware of that and I will compensate for it.

**Comment by nittka** [comment]
If you plan to add further commits, you can mark this PR as draft PR (there is a corresponding link in the "reviewers" section on the left). This is a good indicator for us that you are not done, yet, but feedback can already be given.

**Comment by nittka** [comment]
Due to the number of changes, inline comments are not possible... As far as I can see, the corrections you made to the existing entries are fine. Please remove your additional comments (leave only the correction).
* line 6663 (date changed)
* line 15837 (Correction)
* line 17709,17714 (Error in Translation) - rather a copy/paste error

Line 29506 is OK (beautiful twice) - please remove the comment

@scr0llbaer9000  please check line 54575 imdb tt0751102

**Comment by scr0llbaer9000** [comment]
> line 6663 (date changed)

Just a general remark, in this example, release date, according to Wikipedia: "December 22, 1989 (Italy), April 27, 1990 (United States)"

So it wasn't "wrong", though no problem to change it, but in general, for the future, there may be several examples like this (I think I in my entries there are also some examples where there was a release date for a film festival earlier but I picked the date for the later broad release).

> please check line 54575 imdb tt0751102

It's correct as is. English original episode has a German title to be "exotic". https://en.wikipedia.org/wiki/Die_Hand_Die_Verletzt — and German version didn't use that title because not "exotic" in German. `;)`

**Review by nittka — CHANGES_REQUESTED** [review]
see previous comments; in particular revert the changes around line 54575

**Review by nittka — APPROVED** [review]
@scr0llbaer9000: Do you have objections? Otherwise I would merge.

**Comment by pauldim** [comment]
I'm not finished this file.

**Comment by nittka** [comment]
OK. I converted the PR to a draft. Once, you are done, you can mark it as ready for review.

**Comment by pauldim** [comment]
I couldn't find where it changes.

File: res/database/Default/database_programmes.xml [hunk 1]
```diff
@@ -33508,6 +33539,7 @@ Doborowa obsada.</pl>
 			<title>
 				<de>Longdrink</de>
 				<en>Longdrink</en>
+				<pl>Longdrink - Drink</pl>
```
Comment (nittka) [inline res/database/Default/database_programmes.xml]: intentional?

File: res/database/Default/database_programmes.xml [hunk 2]
```diff
@@ -34508,6 +34574,7 @@ Zawiera wiele kalamburów i [3|Full] w nagrodzonym Oscarem humorze.</pl>
 			<title>
 				<de>Nollywood Chainsaw Bitches - Mit Kettensägen spielt man nicht</de>
 				<en>Nollywood Chainsaw Bitches</en>
+				<pl>Hollywood Chainsaw Bitches</pl>
```
Comment (nittka) [inline res/database/Default/database_programmes.xml]: Nollywood? (Spoofing)

File: res/database/Default/database_programmes.xml [hunk 3]
```diff
@@ -34646,6 +34717,7 @@ Zawiera wiele kalamburów i [3|Full] w nagrodzonym Oscarem humorze.</pl>
 			<title>
 				<de>Hannah - Der Kinofilm</de>
 				<en>Hannah</en>
+				<pl>Baletnica Anna</pl>
```
Comment (nittka) [inline res/database/Default/database_programmes.xml]: ... or did you put the original titles in the "spoofed" title tag.

File: res/database/Default/database_programmes.xml [hunk 4]
```diff
@@ -35776,6 +35886,7 @@ Zawiera wiele kalamburów i [3|Full] w nagrodzonym Oscarem humorze.</pl>
 			<title>
 				<de>[0|First] - Der Extraterr-Friesische</de>
 				<en>[0|First] - The Extraterr-Frisian</en>
+				<pl>[0|First] - Der Extraterr-Friesische</pl>
```
Comment (nittka) [inline res/database/Default/database_programmes.xml]: German version intentional?

File: res/database/Default/database_programmes.xml [hunk 5]
```diff
@@ -35808,6 +35919,7 @@ Zawiera wiele kalamburów i [3|Full] w nagrodzonym Oscarem humorze.</pl>
 			<title>
 				<de>Herbstmolke</de>
 				<en>Autumn Whey</en>
+				<pl>Herbstmolke - Jesienne mleczko</pl>
```
Comment (nittka) [inline res/database/Default/database_programmes.xml]: Just "Jesienne mleczko"?

**Review by nittka — CHANGES_REQUESTED** [review]
In some cases it looks like the real film title was used. But those would have to go into the "title_original" tag. All titles should be "spoofed."

File: res/database/Default/database_programmes.xml [hunk 6]
```diff
@@ -36930,6 +36962,7 @@ Marynarka Wojenna wspierała ten film, niestety nie odniósł on takiego sukcesu
 			<title>
 				<de>An Asian Spooky Story II</de>
 				<en>An Asian Spooky Story II</en>
+				<pl>An Asian Ghost Story II</pl>
```
Comment (nittka) [inline res/database/Default/database_programmes.xml:36965]: If not Polish why change the English version?

**Review by nittka — CHANGES_REQUESTED** [review]

File: res/database/Default/database_programmes.xml (hunk 1 above)
Comment (pauldim) [reply res/database/Default/database_programmes.xml]: Most of the videos I try to give the original and Polish name. In this case, longdrink is the drink in Polish.

File: res/database/Default/database_programmes.xml (hunk 2 above)
Comment (pauldim) [reply res/database/Default/database_programmes.xml]: Hollywood copyrighted? Changed.

File: res/database/Default/database_programmes.xml (hunk 3 above)
Comment (pauldim) [reply res/database/Default/database_programmes.xml]: Changed.

File: res/database/Default/database_programmes.xml (hunk 4 above)
Comment (pauldim) [reply res/database/Default/database_programmes.xml]: A German film is a German title when there is no Polish equivalent.

File: res/database/Default/database_programmes.xml (hunk 5 above)
Comment (pauldim) [reply res/database/Default/database_programmes.xml]: What is molke in German? 
Not enough?

File: res/database/Default/database_programmes.xml [hunk 7]
```diff
@@ -36516,6 +36535,7 @@ Uważany za jeden z najlepszych filmów [0|Full].</pl>
 			<title>
 				<de>Bernie und Brianna auf dem Roten Kontinent</de>
 				<en>The Retriever Mice in Kangaroo Land</en>
+				<pl>Bernard i Bianka w Australii</pl>
```
[the remark this answers is not in the document]
Comment (pauldim) [reply res/database/Default/database_programmes.xml:36538]: changed.

File: res/database/Default/database_programmes.xml (hunk 6 above)
Comment (pauldim) [reply res/database/Default/database_programmes.xml:36965]: changed.

File: res/database/Default/database_programmes.xml (hunk 5 above)
Comment (nittka) [reply res/database/Default/database_programmes.xml]: Molke is the fluid part left over when turning milk into cheese.

Comment (GWRon) [reply res/database/Default/database_programmes.xml]: Think it is serwatka. Maybe circumscribed as "cheese water".

**Comment by scr0llbaer9000** [comment]
> A German film is a German title when there is no Polish equivalent.

Btw you can take liberties in such examples (I did so too) and make up a Polish title also for the `_original' tag. (But I wrote it in the comment when I did so.) The information of the *actual* original title is already there and easily found for the tag of the original language. So you can think about what the Polish title of a film *might* be *if* it got an official localization. I just think it will be a better experience for players who play with the original data.

**Comment by scr0llbaer9000** [comment]
Don't do other files in this PR, as there's already a new PR that may have changes.

**Comment by pauldim** [comment]
> Don't do other files in this PR, as there's already a new PR that may have changes.

OK. Done

**Comment by nittka** [comment]
Are you finished with programme.xml? For the first entries, you added descriptions as well, for the last ones only the title...

**Comment by pauldim** [comment]
> Are you finished with programme.xml? For the first entries, you added descriptions as well, for the last ones only the title...

Finished. File ready for review.

**Comment by scr0llbaer9000** [comment]
A lot of descriptions are still not translations but taken verbatim from sites like https://www.filmweb.pl/

Some of which might still mention original titles of films and original names in clear text. At least these will probably have to be rephrased.

I guess input from @GWRon is needed if that can be allowed, at least for the time being.

**Comment by GWRon** [comment]
We should not add "problems" we did not have before.
- problems can raise from newly added "original titles" ...
- but also from "copy-paste texts" (because descriptions are "lengthier" and thus easier to fall under some kind of "copyright" (in the sense of "Urheberrecht/Schöpfungshöhe")

**Comment by scr0llbaer9000** [comment]
So, Pauldim, do you want to go through the descriptions again and apply the translations?

Sorry that you already put a lot of effort in.

As for the problems with names you've mentioned, maybe you can rephrase or just leave those parts out.

**Comment by pauldim** [comment]
> So, Pauldim, do you want to go through the descriptions again and apply the translations?
> 
> Sorry that you already put a lot of effort in.
> 
> As for the problems with names you've mentioned, maybe you can rephrase or just leave those parts out.

I now propose to merge this file as it is, and I promise to change and improve these descriptions as we add images to the programs. Then you will have to revisit it and change it. Unfortunately, I did not know that you pay so much attention to the program descriptions. As I say, I promise to change it and translate the descriptions into yours. Now you want to add new content, so this blocks further work and yours and mine.

**Comment by nittka** [comment]
I'd be fine with that. @GWRon ?

**Review by nittka — APPROVED** [review]

**Comment by nittka** [comment]
(Please do not hope fore fast progress with respect to custom images. There are several issues that have to be discussed before even starting work: needed framework changes, format for linking existing database entries with the images, place of storage - I don't see the images being part of this repository, etc.)

**Comment by pauldim** [comment]
Then let me know if I should correct these descriptions now. As for the new content, I will translate the files and upload to Scroolbear.

As for the images, I'm a little disappointed. :(

**Comment by GWRon** [comment]
> As for the images, I'm a little disappointed. :(

Not all is lost - it just requires more "preparation" than you might think. It is not removed from our "possibly-todo" :)

**Comment by GWRon** [comment]
> I'd be fine with that. @GWRon ?

I would accept what you guys think is OK ... majority decides (does not mean I am against or so! just clarifying that I do not need to give my "OK" if 2 of you already agreed and it is not something like adding a news about me being a dick :))

**Comment by nittka** [comment]
My suggestion is to wait with further changes until the other PR is merged in order to prevent conflicts.

**Pull request merged**